website/content/en/v0.10.0/getting-started/migrating-from-cas/scripts/step07-tag-security-groups.sh

```diff
@@ -7,7 +7,7 @@ LAUNCH_TEMPLATE=$(aws eks describe-nodegroup --cluster-name ${CLUSTER_NAME} \
 
 SECURITY_GROUPS=$(aws ec2 describe-launch-template-versions \
     --launch-template-id ${LAUNCH_TEMPLATE%,*} --versions ${LAUNCH_TEMPLATE#*,} \
-    --query 'LaunchTemplateVersions[0].LaunchTemplateData.SecurityGroupIds' \
+    --query 'LaunchTemplateVersions[0].LaunchTemplateData.NetworkInterfaces[0].Groups' \
```

Comment:
Oh, good catch. Maybe a simple `if-else` loop to take care of both the situations so that users don't have to worry about verifying the `$SECURITY_GROUPS`. The code looks a bit long but it does the job.

Reply:
I'm guessing there are some other cases where this query won't return what we expect.  I wonder if there is a more consistent way to get this data.

Reply:
[AWS documentation](https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-launch-template-versions.html),  confirms that `SecurityGroupIds` and `Groups` are the only two output fields that return the security group IDs. So, we got it covered! :)

Reply:
But this also assumes a custom LaunchTemplate is used by the Managed Node Group.  In the case that the Node Group is not using a custom Launch Template, we will get nothing in return. 

 I wonder if it would be worth it to instead just get the cluster security group for the guide, then put a disclaimer for cases when your Node Group is using a custom launch template.

Example:
```
SECURITY_GROUPS=$(aws eks describe-cluster --name ${CLUSTER_NAME} --query "cluster.resourcesVpcConfig.clusterSecurityGroupId")
```

Reply:
That would be a good idea, but it may not be convenient for the user. We are also breaking the flow here as the users needs to revisit infrastructure and build commands on their own during installation :) 
Or maybe we can script it for them. Something like -
`SECURITY_GROUPS=$(aws ec2 describe-launch-template-versions \
    --launch-template-id ${LAUNCH_TEMPLATE%,*} --versions ${LAUNCH_TEMPLATE#*,} \
    --query 'LaunchTemplateVersions[0].LaunchTemplateData.SecurityGroupIds' \
    --output text)

if [ $SECURITY_GROUPS == 'None' ]; then
    SECURITY_GROUPS=$(aws ec2 describe-launch-template-versions \
        --launch-template-id ${LAUNCH_TEMPLATE%,*} --versions ${LAUNCH_TEMPLATE#*,} \
        --query 'LaunchTemplateVersions[0].LaunchTemplateData.NetworkInterfaces[0].Groups' \
        --output text)
    if [ $SECURITY_GROUPS == 'None' ]; then
        SECURITY_GROUPS=$(aws eks describe-cluster --name ${CLUSTER_NAME} --query "cluster.resourcesVpcConfig.clusterSecurityGroupId")
    fi
fi`
Will it be a overkill in documentation?

Reply:
I don't think we can account for all possible customer environments in this guide. In fact we are only collecting the security groups for one of the node groups in this guide:

 >"This command only tags the security groups for the first nodegroup in the cluster. If you have multiple nodegroups or multiple security groups you will need to decide which one Karpenter should use."

By getting the Cluster Security Group we are providing a security group which _should_ allow nodes launched by Karpenter register and talk to the control plane.  I suppose we could have 2 sections.  The first, would be to get the Cluster Security Group and then a second section which outlines a couple methods to get the "custom" security groups. 

Would that make sense?

Reply:
Yes, it sure is. Something like this you are suggesting?
--------

When your EKS setup is configured to use only Cluster security group, then please use -

```sh
SECURITY_GROUPS=$(aws eks describe-cluster \
    --name ${CLUSTER_NAME} --query "cluster.resourcesVpcConfig.clusterSecurityGroupId")
```

When your setup uses the security groups in the Launch template of a managed node group, then :

1. For cluster launched using `eksctl` utility:
```sh
SECURITY_GROUPS=$(aws ec2 describe-launch-template-versions \
    --launch-template-id ${LAUNCH_TEMPLATE%,*} --versions ${LAUNCH_TEMPLATE#*,} \
    --query 'LaunchTemplateVersions[0].LaunchTemplateData.SecurityGroupIds' \
    --output text)
```
2. For cluster that is launched by other means (IaC/AWS Console):
```sh
SECURITY_GROUPS=$(aws ec2 describe-launch-template-versions \
    --launch-template-id ${LAUNCH_TEMPLATE%,*} --versions ${LAUNCH_TEMPLATE#*,} \
    --query 'LaunchTemplateVersions[0].LaunchTemplateData.NetworkInterfaces[0].Groups' \
    --output text)
```

Reply:
You could alternatively just || the fields to not specify the means of how the SG is created
```
aws ec2 describe-launch-template-versions \
    --launch-template-id ${LAUNCH_TEMPLATE%,*} --versions ${LAUNCH_TEMPLATE#*,} \
    --query 'LaunchTemplateVersions[0].LaunchTemplateData.[NetworkInterfaces[0].Groups||SecurityGroupIds]' \
    --output text
```

Reply:
@dewjam This looks better? Thanks @njtran for || trick!
------------------------
When your EKS setup is configured to use only Cluster security group, then please use -

```sh
SECURITY_GROUPS=$(aws eks describe-cluster \
    --name ${CLUSTER_NAME} --query "cluster.resourcesVpcConfig.clusterSecurityGroupId")
```

When your setup uses the security groups in the Launch template of a managed node group, then :

```sh
SECURITY_GROUPS=$(aws ec2 describe-launch-template-versions \
    --launch-template-id ${LAUNCH_TEMPLATE%,*} --versions ${LAUNCH_TEMPLATE#*,} \
    --query 'LaunchTemplateVersions[0].LaunchTemplateData.[NetworkInterfaces[0].Groups||SecurityGroupIds]' \
    --output text)
```